src/main/java/hudson/plugins/git/extensions/impl/CloneOption.java

```diff
@@ -148,7 +149,12 @@ public void decorateCloneCommand(GitSCM scm, Run<?, ?> build, GitClient git, Tas
             // in a single job definition.
             RemoteConfig rc = scm.getRepositories().get(0);
             List<RefSpec> refspecs = rc.getFetchRefSpecs();
-            cmd.refspecs(refspecs);
+            List<RefSpec> expandedRefSpecs = new ArrayList<>();
+            EnvVars env = build.getEnvironment(listener);
+            for (RefSpec ref : refspecs) {
```

Comment:
Needs automated tests to confirm:
* Operating system environment variables (like PATH) are expanded as expected (PATH will generate a nonsense refspec, but it has the benefit of being defined on all supported platforms)
* Jenkins variables (like [GIT_BRANCH](https://plugins.jenkins.io/git#branch-variables)) are expanded as expected in a Freestyle project
* Jenkins token macros (like [GIT_REVISION](https://plugins.jenkins.io/token-macro)) are expanded as expected in a Freestyle project

Reply:
I have added a test file which is running successful tests for "BUILD_ID", "BUILD_NO" and "GIT_COMMIT". 
I have two issues here: 

- I can't make a freestyle build with ${PATH} added in the refspec, it fails the build with a "_can't file the commitFile in the working directory_". Although I have tested this **interactively** where you can clearly see that the refspec is expanded but the build fails, I still would want your opinion on this.
<img width="1440" alt="Screenshot 2020-02-07 at 9 19 29 PM" src="https://user-images.githubusercontent.com/31189405/74044272-d43d7e80-49f0-11ea-94ff-40aa78e59435.png">

- By Jenkins token macros do you mean running a build with parameters? Is that how tokens are used? If yes then how does one set a parameter for a job? Should I put the variable in the EnvVars store?

Reply:
I've seen the Token Macro API but I don't think that servers our purpose here 
`assertThat(TokenMacro.expandAll(build2, listener, "${GIT_REVISION}"), is(sha1String));`

Reply:
After doing some RCA I found out that adding ${PATH} to refspec would make it illegal, hence the job will fail before doing anything else.

Reply:
> I can't make a freestyle build with ${PATH} added in the refspec, it fails the build with a "can't file the commitFile in the working directory"

How did you tried? Just adding `${PATH}` to `permuteRefSpecVariable`?

I'd also like to know how this PR behaves with job parameters (https://wiki.jenkins.io/display/JENKINS/Define+Parameters)